tensorboard/plugins/scalar/polymer3/tf_scalar_dashboard/tf-scalar-card.ts

```diff
@@ -0,0 +1,354 @@
+        title="Toggle y-axis log scale"
+      ></paper-icon-button>
+      <paper-icon-button
+        icon="settings-overscan"
+        on-tap="_resetDomain"
+        title="Fit domain to data"
+      ></paper-icon-button>
+      <template is="dom-if" if="[[showDownloadLinks]]">
+        <paper-menu-button on-paper-dropdown-open="_updateDownloadLink">
+          <paper-icon-button
+            class="dropdown-trigger"
+            slot="dropdown-trigger"
+            icon="file-download"
+          ></paper-icon-button>
+          <paper-listbox class="dropdown-content" slot="dropdown-content">
+            <paper-item>
+              <a id="svgLink" download="[[tag]].svg">
+                Download Current Chart as SVG
+              </a>
+            </paper-item>
+          </paper-listbox>
+        </paper-menu-button>
+      </template>
+      <span style="flex-grow: 1"></span>
+      <template is="dom-if" if="[[showDownloadLinks]]">
+        <div class="download-links">
+          <tf-downloader
+            runs="[[_runsFromData(dataToLoad)]]"
+            tag="[[tag]]"
+            url-fn="[[_downloadUrlFn]]"
+          ></tf-downloader>
+        </div>
+      </template>
+    </div>
+    <style>
+      :host {
+        margin: 5px;
+        display: block;
+        width: 330px;
+      }
+
+      :host([_expanded]) {
+        width: 100%;
+      }
+
+      :host([_expanded]) #tf-line-chart-data-loader-container {
+        height: 400px;
+      }
+
+      #tf-line-chart-data-loader-container {
+        height: 200px;
+        width: 100%;
+      }
+
+      tf-card-heading {
+        display: block;
+        margin-bottom: 10px;
+      }
+
+      #buttons {
+        display: flex;
+        flex-direction: row;
+      }
+
+      paper-icon-button {
+        color: #2196f3;
+        border-radius: 100%;
+        width: 32px;
+        height: 32px;
+        padding: 4px;
+      }
+
+      paper-icon-button[selected] {
+        background: var(--tb-ui-light-accent);
+      }
+
+      .download-links {
+        display: flex;
+        height: 32px;
+      }
+
+      .download-links a {
+        align-self: center;
+        font-size: 10px;
+        margin: 2px;
+      }
+
+      .download-links paper-dropdown-menu {
+        width: 100px;
+        --paper-input-container-label: {
+          font-size: 10px;
+        }
+        --paper-input-container-input: {
+          font-size: 10px;
+        }
+      }
+
+      paper-menu-button {
+        padding: 0;
+      }
+      paper-item a {
+        color: inherit;
+        text-decoration: none;
+        white-space: nowrap;
+      }
+    </style>
+  `;
+
+  @property({type: String})
+  tag: string;
+
+  @property({type: Array})
+  dataToLoad: object[];
+
+  @property({type: String})
+  xType: string;
+
+  @property({type: Boolean})
+  active: boolean;
+
+  @property({type: Boolean})
+  ignoreYOutliers: boolean;
+
+  @property({type: Object})
+  requestManager: object;
```

Comment:
Can we type this as `RequestManager`? I’ve generally been trying to
include the appropriate types where possible; okay with me to drop it
for something like `tagMetadata` if you can’t be bothered, but since
`RequestManager` is a pure-TypeScript class anyway it seems worth imho.

Reply:
Done, just overlooked it.